test/v2/connectors/V2BaseConnectorSpec.scala

```diff
@@ -300,6 +300,102 @@ class V2BaseConnectorSpec
         verify(sut, times(1)).setHeader(ArgumentMatchers.eq(Seq((HeaderNames.AUTHORIZATION -> token))): _*)
     }
 
+    "withMovementId adds the audit movement Id header for movement value" in forAll(arbitrary[MovementId]) {
+      movementId =>
+        val sut = mock[RequestBuilder]
+        // any here, verify later
+        when(sut.setHeader(any())).thenReturn(sut)
+
+        sut.withMovementId(Some(movementId))
+        verify(sut, times(1)).setHeader(ArgumentMatchers.eq(Seq(("X-Audit-Meta-Movement-Id" -> movementId.value))): _*)
+    }
+
+    "withMovementId ignore the audit movement Id header for None value" in {
+      val sut = mock[RequestBuilder]
+      // any here, verify later
+      when(sut.setHeader(any())).thenReturn(sut)
+
+      sut.withMovementId(None)
+      verify(sut, times(0)).setHeader(ArgumentMatchers.eq(Seq(("X-Audit-Meta-Movement-Id" -> "None"))): _*)
```

Comment:
That's not quite right -- we want to verify it's not called at all, no matter the arguments. All you're verifying here is that this isn't called with this specific argument list, which is true, but

```scala
        verify(sut, times(1)).setHeader(ArgumentMatchers.eq(Seq(("X-Audit-Meta-Movement-Id" -> movementId.value))): _*)
```

could still be true.

Something like

```scala
        verify(sut, times(0)).setHeader(ArgumentMatchers.eq(any()))
```

is what you really want.

Reply:
It's not true here specifically, but It _could_ be true if you introduced that as a bug in the code.

I guess the problem is that it's a variardic parameter. I don't know how to work around it, but I don't have the answers to everything. I just want the tests to be properly representative of what happens.

You might be able to just use `verifyNoInteractions` (or whatever it's called) instead.